### ISSUE TITLE
"write error: Broken pipe" in terminal

### ISSUE THREAD

**Issue by rlipscombe** [microsoft/vscode#26961]
- VSCode Version: 1.12.2
- OS Version: Linux Mint 17.3 (Ubuntu 14.04 LTS)

Steps to Reproduce:

1. Open the terminal, running /bin/bash
2. Paste in the code for the _random_string function, shown below.
3. Run the function. You'll get "write error: Broken pipe"
4. Note that this does not happen in gnome-terminal on the same PC.

```
function _random_string() {
    env LC_CTYPE=C tr -dc -- 'A-Za-z0-9' < /dev/urandom | head -c 12
}
```

Expected output (per gnome-terminal):

```
roger@roger-pc:~ $ function _random_string() { env LC_CTYPE=C tr -dc -- 'A-Za-z0-9' < /dev/urandom | head -c 12; }
roger@roger-pc:~ $ _random_string
Es3YEHmzezgCroger@roger-pc:~ $ 
```

Actual output (in VS code terminal):

```
roger@roger-pc:~ $ function _random_string() { env LC_CTYPE=C tr -dc -- 'A-Za-z0-9' < /dev/urandom | head -c 12; }
roger@roger-pc:~ $ _random_string
92wcrvYW2Cx8tr: write error: Broken pipe
tr: write error
```

**Comment by rlipscombe** [comment]
The behaviour in VS Code on Mac is differently broken. The command outputs the expected random string, and then hangs. According to the dropdown in the top-right of the integrated terminal, it's still running `tr`.

The behaviour in iTerm2 is correct (i.e. the same as gnome-terminal).

**Comment by rlipscombe** [comment]
The behaviour in VS Code on Windows (Creator's update), when using WSL Bash, is correct.

**Comment by GIThunte** [comment]
Hi, I installed the vc code, there is also such an error, see the video
https://www.youtube.com/watch?v=HJKO7kLMrwc&feature=youtu.be

Version 1.19.1
Commit 0759f77bb8d86658bc935a10a64f6182c5a1eeba
Date 2017-12-19T09:41:01.414Z
Shell 1.7.9
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64

**Comment by rlipscombe** [comment]
Still broken in 1.19.2 on Linux Mint 18.3:

```
$ tr -dc -- 'A' < /dev/urandom | head -c 12
AAAAAAAAAAAAtr: write error: Broken pipe
tr: write error
```

```
$ cat /dev/urandom | tr -dc -- 'A' | head -c 12
AAAAAAAAAAAAtr: write error: Broken pipe
tr: write error
cat: write error: Broken pipe
```

It's got something to do with the `</dev/urandom`, because the following works fine:

```
$ echo 'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' | tr -dc -- 'A' | head -c 12
AAAAAAAAAAAA
```

----

Version 1.19.2
Commit 490ef761b76b3f3b3832eff7a588aac891e5fe80
Date 2018-01-10T15:57:21.071Z
Shell 1.7.9
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64

**Comment by Tyriar** [comment]
https://github.com/Microsoft/vscode/issues/26961#issuecomment-357001068

**Comment by GIThunte** [comment]
I still do not work, I do not want to put a newer system only because of an error in the vs code

========================================================

who@v:~$ cat /dev/urandom | tr -dc -- 'A' | head -c 12
AAAAAAAAAAAAtr: write error: Broken pipe
tr: write error
cat: write error: Broken pipe



who@v:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.3 LTS
Release:        16.04
Codename:       xenial


========================================================

**Comment by thedrumchannell** [comment]
I am seeing this error in Windows when running a make command.

Version 1.19.3

make: write error
The terminal process terminated with exit code: 1

**Comment by kbumsik** [comment]
In the current v1.24.1 and Insider v1.25,
`cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 8 | head -n 1` does not exit.
also `yes | head -n 1` outputs `yes: standard output: Broken pipe`.

**Comment by Tyriar** [comment]
I can no longer repro this on any platform.

**Comment by rlipscombe** [comment]
Confirmed fixed for me in 1.38.0 on Linux.
